libs/sdk-core/src/models.rs

```diff
@@ -132,41 +131,84 @@ pub struct ReverseSwapPairInfo {
 }
 
 /// Details of past or ongoing reverse swaps, as stored in the Breez local DB
-#[derive(Clone, Debug, Serialize)]
+#[derive(Clone, Debug, Eq, PartialEq, Serialize)]
 pub struct ReverseSwapInfo {
-    // static immutable data
+    pub created_at: i64,
+
     pub claim_address: String,
     pub lockup_address: String,
-
-    // dynamic data
-    pub status: ReverseSwapStatus,
+    pub hodl_bolt11: String,
+    pub hodl_preimage: Vec<u8>,
+
+    pub hodl_invoice_paid: bool,
+    pub hodl_invoice_last_payment_attempt_error: Option<String>,
+    pub lockup_tx_id: Option<String>,
+    pub lockup_tx_confirmed: bool,
+    pub claim_tx_id: Option<String>,
+    pub claim_tx_confirmed: bool,
+    // TODO swap amount, lock amount, claimed amount
+
+    // Fields from the Boltz API response
+    /// If the reverse swap creation fails with a Boltz error, the error message is documented here
+    pub boltz_api_create_error: Option<String>,
+    pub hash_info_id: String,
+    pub onchain_amount_sat: u64,
+    pub redeem_script: String,
+    pub timeout_block_height: u32,
+}
+
+impl ReverseSwapInfo {
+    /// Dynamically determine the status, based on the current reverse swap fields, at a specific blockheight
+    pub fn status(&self, blockheight: u32) -> ReverseSwapStatus {
+        match self.boltz_api_create_error {
+            Some(_) => ReverseSwapStatus::InitialError,
+            None => match blockheight >= self.timeout_block_height {
+                true => ReverseSwapStatus::Expired,
+                false => match self.hodl_invoice_paid {
+                    false => match self.hodl_invoice_last_payment_attempt_error {
+                        None => ReverseSwapStatus::Initial,
+                        Some(_) => ReverseSwapStatus::FailedToPayInvoice,
+                    },
+                    true => match self.lockup_tx_confirmed {
+                        false => ReverseSwapStatus::InvoicePaid,
+                        true => match self.claim_tx_confirmed {
+                            false => ReverseSwapStatus::LockTxConfirmed,
+                            true => ReverseSwapStatus::ClaimTxConfirmed,
+                        },
+                    },
+                },
+            },
+        }
+    }
 }
 
 /// The status of a reverse swap
 #[derive(Clone, PartialEq, Eq, Debug, Serialize)]
 pub enum ReverseSwapStatus {
-    /// The reverse swap has been created. The HODL invoice is not yet paid.
+    /// HODL invoice payment is not completed yet
     Initial = 0,
 
+    /// Boltz API failed to create the reverse swap
+    InitialError = 1,
+
     /// The HODL invoice has been paid.
-    InvoicePaid = 1,
+    InvoicePaid = 2,
+
+    /// The reverse swap has been created. Paying the HODL invoice was attempted, but failed.
+    FailedToPayInvoice = 3,
```

Comment:
@ok300  Is there a difference in behavior between FailedToPayInvoice and InitialError?
If yes then two fields are required, if no perhaps it is better to have only one field for error?

Reply:
Good point, got rid of `InitialError`.